xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -900,7 +900,10 @@ bool CGUIMediaWindow::OnClick(int iItem, const std::string &player)
     else if (!CProfilesManager::GetInstance().GetCurrentProfile().canWriteSources() && !g_passwordManager.IsProfileLockUnlocked())
       return false;
 
-    return OnAddMediaSource();
+    if (OnAddMediaSource())
+      Refresh();
+
+    return true;
```

Comment:
To see those things you added. Either clear it or update it I guess, I really don't have an opinion which. But if this change is to ensure that the GUI is updated with the __current__ list of items when a source is added then the cache needs to be considered too. At the moment it works by accident in that the source list isn't always cached.